### DISCUSSION TITLE
Suricata Alerts to syslog

### DISCUSSION THREAD

**Discussion by WPTechnician** [Security-Onion-Solutions/securityonion#2066] [Unsupported Versions] [answered]
I need to send Suricata alerts to an external collector but there isn't much in the way of documentation on how to do that with SO 2.3.  Is there any other documentation or would someone be able to give me the basics on how to get that configured?  Thanks!

**Comment by TOoSmOotH** [comment, marked as answer]
https://docs.securityonion.net/en/2.3/syslog-output.html?highlight=syslog

You will need to add a custom pipeline and clone the IDS events to send to both SOC and your external system.

**Reply by WPTechnician** [reply]
Yes, I did read that in the docs but I am not altogether sure how to go about that.  I am not really conversant with what the syntax of that would be.  I have started a custom file (/opt/so/saltstack/local/salt/logstash/pipelines/config/custom/9999_output_custom.jinja) and have added the following to it:

`Input {
}
filter {
      clone {
      }
    }
Output {
  tcp {
    id => "Out_to_SIEM"
    host => "10.10.10.2"
    port => "1468"
  }
}
`

Thanks!

**Reply by WPTechnician** [reply]
> https://docs.securityonion.net/en/2.3/syslog-output.html?highlight=syslog
> 
> You will need to add a custom pipeline and clone the IDS events to send to both SOC and your external system.

Would you at least give me a clue on the input section?  I can't find any examples anywhere on how to do this and just need a starting point.  I can work through the filter and output but would really appreciate a helping hand on the input.  Thanks!

**Reply by weslambert** [reply]
Try adding a tag in your clone filter and key off of that to send it.  This is really more of an Elastic-related question than Security Onion itself, so you may also want to post in the Elastic forum(s).

**Reply by weslambert** [reply]
You don't really need an input section, as long as the events are already flowing through the pipeline -- you would just need to clone them after they have been typed as `module:suricata`/`event.module:suricata` AND `dataset:alert`/`event.dataset:alert`, depending on where in the pipeline you are placing these.

**Reply by TOoSmOotH** [reply]
https://medium.com/@alhaynes/logstash-clone-filter-547421aabf6d

**Reply by WPTechnician** [reply]
That's a great help, thank you very much!  I will be good from there!

**Reply by WPTechnician** [reply]
So I have events cloning now but it only seems to be records from Zeek despite this being the filter:

![image](https://user-images.githubusercontent.com/19175877/100258601-4302ae80-2f15-11eb-98d5-5119a645d335.png)

I have this configuration in this file:

![image](https://user-images.githubusercontent.com/19175877/100258672-54e45180-2f15-11eb-932f-66cbd724fe5d.png)

Should I have it in a different spot?  Thanks!

**Reply by WPTechnician** [reply]
Ugh, now I just feel dumb.  Thanks for your help, I have it all sorted out now.  Thanks!
